### DISCUSSION TITLE
Are the docs updated

### DISCUSSION THREAD

**Discussion by ThatXliner** [supabase/supabase#9419] [Questions] [answered]
https://supabase.com/docs/guides/api/generating-types#update-types-automatically-with-github-actions

mentions openapi-typescript

but `supabase gen types typescript` exists (now, because I don't remember this existing before).

What's the recommended method?

**Comment by ThatXliner** [comment]
It appears that they are not the same. What is this new command then?

**Reply by vavdav** [reply]
It is a command coming with v2, but it is not documented very well.

https://supabase.com/docs/reference/javascript/next/typescript-support#generating-types

**Comment by soedirgo** [comment, marked as answer]
Hey folks, we've updated the [guide](https://supabase.com/docs/guides/api/generating-types) to use the CLI to generate types. `openapi-typescript` is now only used for the v1 of the JS library. Hope that clears it up - sorry for the confusion!
